src/Components/CategoryInput.tsx

```diff
@@ -11,20 +11,26 @@ interface Props {
 }
 
 export const CategoryInput = ({ getCategory }: Props) => {
-  const inputRef = useRef<string>('')
-  //Should get category data from db
+  const [inputCat, setInputCat] = useState<string>('')
   const [categories, setCategories] = useState<string[]>(['Common misconception', 'Form similar to answer'])
   const [selections, setSelections] = useState<string[]>([])
 
   function addElement() {
-    if (!categories.includes(inputRef.current) && inputRef.current != '')
-      setCategories([inputRef.current, ...categories])
-    else if (inputRef.current == '') alert('Please enter the category') //no input case
+    if (!categories.includes(inputCat) && inputCat != '') setCategories([inputCat, ...categories])
+    else if (inputCat == '') alert('Please enter the category') //no input case
     else alert('You have the same category') //duplicates case
   }
 
-  function removeElement(arr: string[], item: string) {
-    return arr.filter(e => e != item)
+  function selectElement(item: string) {
+    if (selections.includes(item)) {
+      setSelections(selections.filter(s => s != item))
+    } else setSelections([...selections, item])
+    getCategory(selections)
+  }
+
+  function deleteElement(item: string) {
```

Comment:
it is better to use `useCallback` hook when defining functions in a component!

```typescript
const deleteElement = useCallback((item: string) => {
    setCategories(categories.filter(c => c != item))
    setSelections(selections.filter(s => s != item))
}, [setSelections, selections.filter])
```

Reply:
okay.! Actually I don't understand the `useCallback` fully, so I should study it more.